pkgs/mcaptcha/default.nix

```
@@ -0,0 +1,180 @@
+        yarn --offline build
+      '';
+
+      doCheck = true;
+      checkPhase = ''
+        CI=true yarn test
+      '';
+
+      # Copied from `make frontend` in the mCaptcha Makefile:
+      # https://github.com/mCaptcha/mCaptcha/blob/f337ee0643d88723776e1de4e5588dfdb6c0c574/Makefile#L130-L147
+      installPhase = ''
+        runHook preInstall
+
+        yarn run sass -s \
+          compressed templates/main.scss  \
+          ./static/cache/bundle/css/main.css
+        yarn run sass -s \
+          compressed templates/mobile.scss  \
+          ./static/cache/bundle/css/mobile.css
+        yarn run sass -s \
+          compressed templates/widget/main.scss  \
+          ./static/cache/bundle/css/widget.css
+
+        patchShebangs deps/vanilla/scripts
+        deps/vanilla/scripts/librejs.sh
+        deps/vanilla/scripts/cachebust.sh
+
+        mv deps/vanilla/static/cache/bundle $out
+
+        runHook postInstall
+      '';
+
+      # Note that "true" disables the dist phase, as this is all handled in
+      # installPhase above.
+      distPhase = "true";
+    };
+
+  openapi = let
+    src = rootSrc + "/docs/openapi";
+  in
+    mkYarnPackage {
+      inherit src;
+
+      offlineCache = fetchYarnDeps {
+        yarnLock = src + "/yarn.lock";
+        sha256 = "sha256-mdd5AwO4WO/RoR/ycR+miJvRXgu1K6WXvMZtyyV/0Tc=";
+      };
+
+      buildPhase = ''
+        export HOME=$(mktemp -d)
+        yarn --offline build
+      '';
+
+      installPhase = ''
+        runHook preInstall
+        mv deps/mcaptcha/dist $out
+        runHook postInstall
+      '';
+
+      # Note that "true" disables the dist phase, as this is all handled in
+      # installPhase above.
+      distPhase = "true";
+    };
+
+  cache-bust = let
+    src = rootSrc + "/utils/cache-bust";
+  in
+    rustPlatform.buildRustPackage {
+      inherit src;
+      pname = "cache-bust";
+      version = "unstable-${releaseDate}";
+
+      cargoLock = {
+        lockFile = src + "/Cargo.lock";
+        outputHashes = {
+          "cache-buster-0.2.0" = "sha256-FT+GV2c+jZyVyC2pEtDm52Aurbf6tOZeBqzHq3NZptw=";
+        };
+      };
+    };
+
+  mcaptcha = let
+    src = rootSrc;
+  in
+    rustPlatform.buildRustPackage {
+      inherit src;
+      pname = "mcaptcha";
+      version = "unstable-${releaseDate}";
+
+      cargoLock = {
+        lockFile = src + "/Cargo.lock";
+        outputHashes = {
+          "actix-auth-middleware-0.2.0" = "sha256-sLd2Fsa02bXE+CTzTcByTF2PAnzn5YEYGekCmw+AG4E=";
+          "actix-web-codegen-4.0.0" = "sha256-2MKgeCa9C5WL0TtvQSTvz2YMBBgzn7tnkFL7c7KJFSs=";
+          "argon2-creds-0.2.2" = "sha256-A5xkcVvi+xfdQ0vBdqJgtlIbiNmOz6weSB3ho6kAz+A=";
+          "cache-buster-0.2.0" = "sha256-FT+GV2c+jZyVyC2pEtDm52Aurbf6tOZeBqzHq3NZptw=";
+          "libmcaptcha-0.2.3" = "sha256-oedAXrasZ3YAj6PiscdasAQ0RlG9YcFAuFRtxicmkhY=";
+          "pow_sha256-0.3.1" = "sha256-gprbSYL0tjKQlQe/lJwFZM0avSQT92nAwUnr61t0X0g=";
+        };
+      };
+
+      patches = [
```

You can easily get a patch from your PR:

```suggestion
      patches = [
        (fetchpatch {
          url = "https://github.com/mCaptcha/libmcaptcha/pull/13.patch";
          hash = lib.fakeHash;
        })
```

(Of course, add `fetchpatch` as an argument.)

---

We opted to vendor the patch. Thanks for the suggestion! I knew about appending `.patch` to individual commit urls, but I never knew you could do it to a whole commit range.